### PR TITLE
scripts: tests: twister_blackbox: Add test test_report.py

### PR DESCRIPTION
**Pull request by KamilxPaszkiet**
add tests to test_report.py:
- platform_reports
- report_suffix
- report_name
- report_dir
- outdir

### CODE REVIEW

**Review by hakehuang — DISMISSED** [review]

File: scripts/tests/twister_blackbox/test_report.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+                pytest.raises(SystemExit) as sys_exit:
+            self.loader.exec_module(self.twister_module)
+
+        out, err = capfd.readouterr()
+        sys.stdout.write(out)
+        sys.stderr.write(err)
+
+        for f_name in file_name:
+            path = os.path.join(twister_path, f_name)
+            assert os.path.exists(path), 'file not found'
```
Comment (LukaszMrugala) [inline scripts/tests/twister_blackbox/test_report.py:295]: Could use `f_name` in the assert str.

File: scripts/tests/twister_blackbox/test_report.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
```
Comment (LukaszMrugala) [inline scripts/tests/twister_blackbox/test_report.py:263]: Could use `f_name` in the assert str.

**Review by LukaszMrugala — DISMISSED** [review]

File: scripts/tests/twister_blackbox/test_report.py (hunk 1 above)
Comment (KamilxPaszkiet) [reply scripts/tests/twister_blackbox/test_report.py]: fixed

File: scripts/tests/twister_blackbox/test_report.py (hunk 2 above)
Comment (KamilxPaszkiet) [reply scripts/tests/twister_blackbox/test_report.py]: fixed

**Review by nashif — APPROVED** [review]

**Review by LukaszMrugala — APPROVED** [review]

**Pull request merged**